src/Build/Definition/ProjectCollection.cs

```diff
@@ -609,12 +623,9 @@ public ICollection<ILogger> Loggers
             [DebuggerStepThrough]
             get
             {
-                using (_locker.EnterDisposableUpgradeableReadLock())
-                {
-                    return _loggingService.Loggers == null
-                        ? (ICollection<ILogger>)ReadOnlyEmptyCollection<ILogger>.Instance
-                        : new List<ILogger>(_loggingService.Loggers);
-                }
+                return _loggingService.Loggers == null
```

Comment:
`_loggingService` is mutated only as part of `Dispose` so no need to synchronize on that. `Loggers` exposes an unprotected list though, so this needs to be fixed to happen under the `LoggingService`-internal lock.